corehq/apps/app_execution/data_model.py

```diff
@@ -98,11 +104,51 @@ def __str__(self):
         return f"Entity Select: {self.value}"
 
 
+@define
+class QueryInputValidationStep(Step):
+    type: ClassVar[str] = "query_input_validation"
+    is_form_step: ClassVar[bool] = False
+
+    inputs: dict
+    """Search inputs dict. Keys are field names and values are search values."""
+
+    def get_request_data(self, session, data):
+        query_key = session.data["queryKey"]
+        return {
+            **data,
+            # DataDog tag value
+            "requestInitiatedByTag": "field_change",
+            "query_data": {
+                query_key: {
+                    "execute": False,
+                    "force_manual_search": True,
+                    "inputs": self.inputs,
+                }
+            },
+        }
+
+
 @define
 class QueryStep(Step):
     type: ClassVar[str] = "query"
     is_form_step: ClassVar[bool] = False
+
     inputs: dict
+    """Search inputs dict. Keys are field names and values are search values."""
```

Comment:
I haven't seen this pattern before - I assume these docstrings correspond to the vars above them?  Is this an `attrs` pattern?  A newer version of `":param inputs: blah blah"`?

Reply:
It's not an attrs pattern, just something I've seen in other codebases. There was a [PEP](https://peps.python.org/pep-0224/) for it but it was rejected but still some ongoing discussion online: https://discuss.python.org/t/revisiting-attribute-docstrings/36413

Reply:
Oh neat, I see it's [supported by Sphinx already](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoattribute).  I also like the doc comment syntax - seems like it'd be super useful if we used autodoc more.

```python
class Foo:
    """Docstring for class Foo."""

    #: Doc comment for class attribute Foo.bar.
    #: It can have multiple lines.
    bar = 1

    flox = 1.5   #: Doc comment for Foo.flox. One line only.

    baz = 2
    """Docstring for class attribute Foo.baz."""

    def __init__(self):
        #: Doc comment for instance attribute qux.
        self.qux = 3

        self.spam = 4
        """Docstring for instance attribute spam."""
```